### PR TITLE
Copy edit async clipboard article.

### PR DESCRIPTION
**Pull request by jpmedley**
Copy edit pursuant to https://github.com/GoogleChrome/web.dev/issues/1278 of the web.dev migration plan.

### CODE REVIEW

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 1]
```diff
@@ -14,14 +14,14 @@ description: Starting in Chrome 76, the async clipboard API now handles (some) i
...
-`image/png` images.
+of the [Asynchronous Clipboard
+API](https://w3c.github.io/clipboard-apis/#async-clipboard-api). Now in Chrome
+76, adding support for images makes it easy to programmatically copy and paste
+png images.
 
 <aside class="caution">
   <b>Note:</b>
-  At the time of writing, only <code>image/png</code> files are supported.
+  At the time of writing, only png files are supported.
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: When referring generically to a file type don't use the MIME type. 

Is this note still true?

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 2]
```diff
@@ -49,10 +53,10 @@ async function copyPageURL() {
 }
 ```
 
-### Paste: Reading text from the clipboard {: #reading-text }
+### Paste: reading text from the clipboard {: #reading-text }
 
 Much like copy, text can be read from the clipboard by calling
-`navigator.clipboard.readText()` and waiting for the returned Promise to
+`navigator.clipboard.readText()` and waiting for the returned promise to
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: When referring generically to Promises, capitalize it. When referring to a specific instance, don't.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 3]
```diff
@@ -85,18 +89,18 @@ document.addEventListener('paste', async (e) => {
 
 ### Security and permissions {: #security-permission }
 
-The `navigator.clipboard` API is only supported for pages served over HTTPS,
+The `navigator.clipboard` property is only supported for pages served over HTTPS,
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Reserve API for whole specs or sets of interfaces. In general you should be specific about whether you talking about a property, method, event, whatever.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 4]
```diff
@@ -85,18 +89,18 @@ document.addEventListener('paste', async (e) => {
...
 and to help prevent abuse, clipboard access is only allowed when a page is
 the active tab. Pages in active tabs can write to the clipboard without
 requesting permission, but reading from the clipboard always requires
 permission.
 
 When the Asynchronous Clipboard API was introduced,
 two new permissions for copy and paste were added to the
-[Permissions API](/web/updates/2015/04/permissions-api-for-the-web):
+ [Permissions API](/web/updates/2015/04/permissions-api-for-the-web):
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: The rendered version needed a space between this and the previous word.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 5]
```diff
@@ -112,37 +116,40 @@ permissionStatus.onchange = () => {
 };
 ```
 
-## 🆕 The new image-focused portion of the Asynchronous Clipboard API {: #images }
+## The new image-focused portion of the Asynchronous Clipboard API {: #images }
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: I think this title is wordy, but I don't have another idea.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 6]
```diff
@@ -112,37 +116,40 @@ permissionStatus.onchange = () => {
...
+To write an image to the clipboard, you need the image as a
+[`Blob`][blob]. One way to do
+this is by requesting the image from an server by calling `fetch()` (or
+`XMLHTTPReuest()`). The `response` object returned by `fetch()` has a [`blob()`
+method][blob-method] and
+`XMLHTTPRequest()` let's you set `"blob"` as the [`responseType`][blog-response-type]. 
+
+Calling the server may not be desireable or possible for a variety of reasons.
+Fortunately, you can also write the image to a canvas, then call
+[`HTMLCanvasElement.toBlog()`][to-blob].
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: I reworked this whole paragraph to avoid using API keywords as different parts of speech. Changing API keywords also requires mixing fonts in the same word, which should also not be done.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 7]
```diff
@@ -112,37 +116,40 @@ permissionStatus.onchange = () => {
...
+[blob-method]: https://developer.mozilla.org/en-US/docs/Web/API/Body/blob
 [to-blob]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob
 [blog-response-type]: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType#Value
 
-Next, pass an array of `ClipboardItem`s as a parameter to the `write()` method.
+Next, pass an array of `ClipboardItem` objects as a parameter to the `write()` method.
 Currently you can only pass one image at a time, but we plan to add support for
 multiple images in the future.
 
 The `ClipboardItem` takes an object with the MIME type of the image as the key,
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: I don't understand this at all. I tried to find MIME types in the creation of Clipboard instances in the spec and couldn't.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 8]
```diff
@@ -112,37 +116,40 @@ permissionStatus.onchange = () => {
...
 and the actual blob as the value. The sample code below shows a future-proof
-way to do this by leveraging the [`Object.defineProperty()`][object-define-prop]
-method. Using this approach will ensure your code will be ready for future
-image types as well as other MIME types that the Asynchronous Clipboard API
-may support.
+way to do this by using the [`Object.defineProperty()`][object-define-prop]
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: 'leveraging' to 'using'

'Leveraging' means 'to force'. You're not forcing anything here. You're simply using it.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 9]
```diff
@@ -203,8 +210,8 @@ async function getClipboardContents() {
 
 ### Custom paste handler {: #custom-paste-handler }
 
-If you want to dynamically handle paste events, you can listen for the `paste`
-event, [prevent the default][prevent-default] behavior, then use the code above
+To dynamically handle paste events, listen for the `paste`
+event, call [`preventDefault()`][prevent-default], then use the code above
```
Comment (jpmedley) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Why, specifically, is `preventDefault()` being called? I'm not able to infer it. It might be something we need to explain.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md [hunk 10]
```diff
@@ -31,12 +31,16 @@ API works. If you remember the details, skip ahead to the
 
 ## Recap of the Asynchronous Clipboard API {: #recap }
 
-### Copy: Writing text to the clipboard {: #copy-text }
+Before describing image support, I want to review how the Asynchronous Clipboard
+API works. Feel free to [skip ahead](#images) if you're already comforntable
```
Comment (petele) [inline src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: spelling of comfortable

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 5 above)
Comment (petele) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: As you said in another PR - it won't be new forever ;)

**Comment by petele** [comment]
This was @tomayac's article, he should review.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 1 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: This is still true, yes.

Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Oh, and thanks for the style note!

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 2 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Thanks for the style note.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 3 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Gotcha.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 5 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: The—at the time of writing—new image-focused portion of the Asynchronous Clipboard API 🤣

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 7 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: This refers to this step in the code:

```js
await navigator.clipboard.write([
    new ClipboardItem(Object.defineProperty({}, blob.type, {
      value: blob,
      enumerable: true
    }))
  ]);
```

Me, too(?), I think the API is a bit wordy here, but I believe it's like that because eventually it will be opened to all MIME types, meaning it needs full flexibility. I can imagine there will be convenience methods introduced in the future for more common clipboard tasks like fiddling with images (text already has a convenience method, as described in the article).

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 8 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: TIL this subtlety. I was leveraging the word leverage all wrong for my entire life. Pun intended.  Thanks for teaching me English!

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 9 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Oh, simply because you might want to do anything else before actually pasting, like maybe crop to square and zoom on face if you work on something like an avatar picker or so.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 10 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Fixed.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 7 above)
Comment (jpmedley) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: I still don't see a MIME type.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 9 above)
Comment (jpmedley) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: That is certainly worth mentioning in the text, and not something I'd assume readers would just think of.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 6 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Thanks!

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 7 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: It's the `blob.type` part in the code.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 9 above)
Comment (tomayac) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: Added a clarification.

File: src/content/en/updates/2019/07/image-support-for-async-clipboard.md (hunk 7 above)
Comment (jpmedley) [reply src/content/en/updates/2019/07/image-support-for-async-clipboard.md]: @tomayac I've added a line below that I think clarifies the matter. Please take a look.
